media_kit/lib/src/player/native/player/real.dart

```diff
@@ -1343,6 +1343,71 @@ class NativePlayer extends PlatformPlayer {
     calloc.free(data);
   }
 
+  /// Observes property for the internal libmpv instance of this [Player].
+  /// Please use this method only if you know what you are doing, existing methods in [Player] implementation are suited for the most use cases.
+  ///
+  /// See:
+  /// * https://mpv.io/manual/master/#options
+  /// * https://mpv.io/manual/master/#properties
+  ///
+  Future<int> observeProperty(
+    String property,
+    Future<void> Function(String) listener,
+  ) async {
+    if (disposed) {
+      throw AssertionError('[Player] has been disposed');
+    }
+    await waitForPlayerInitialization;
+    await waitForVideoControllerInitializationIfAttached;
+
+    if (observed.containsKey(property)) {
+      throw ArgumentError.value(
+        property,
+        'property',
+        'Already observed',
+      );
+    }
+    final reply = Random().nextInt(1 << 16) + 1;
```

Comment:
I'm going to be the boring one (again) ^^

I'm not a fan of random here, the risk of collision exists, even if it's very small, especially as it's a "hidden" API.

Instead, I would have used the `hashCode` of the property name.
Knowing that the number of properties is relatively low, the probability of collision is even lower.
The other advantage is that if a collision happens, it won't happen randomly but all the time, which will be easier to debug.

In short, it's a small detail, but one that can be a real pain to debug the day it happens (true story).

Reply:
That's awesome feedback.

I don't know why I spent so much time thinking about saving additional `int`. This approach eliminates the need for `ObservedProperty` class entirely. Everything redundant.

The `hashCode` indeed works great for this purpose.